client/sttp-client4/src/main/scala/sttp/tapir/client/sttp4/SttpClientInterpreter.scala

```diff
@@ -0,0 +1,19 @@
+package sttp.tapir.client.sttp4
+
+import sttp.tapir.{DecodeResult, Endpoint, PublicEndpoint}
+
+private[sttp4] trait SttpClientInterpreter {
+  final protected def throwDecodeFailures[T](dr: DecodeResult[T]): T =
+    dr match {
+      case DecodeResult.Value(v)    => v
+      case DecodeResult.Error(_, e) => throw e
```

Comment:
I'm wondering if we shouldn't use sttp's `DeserializationException` here, which would preserve the metadata, and allow metric backends to report metrics properly?

Reply:
I think I am not sure what to pass in the response field. Could you give me a hand here please?